packages/react-cards/src/components/Card/Card.view.tsx

```diff
@@ -1,19 +1,76 @@
 /** @jsx withSlots */
+import * as React from 'react';
 import { withSlots, getSlots } from '@uifabric/foundation';
-import { getNativeProps, htmlElementProperties } from '@uifabric/utilities';
+import { getNativeProps, htmlElementProperties, warn } from '@uifabric/utilities';
 import { Stack } from 'office-ui-fabric-react/lib/Stack';
 
-import { ICardComponent, ICardProps, ICardSlots } from './Card.types';
+import { ICardComponent, ICardProps, ICardSlots, ICardTokens } from './Card.types';
+import { CardItem } from './CardItem/CardItem';
+import { ICardItemProps } from './CardItem/CardItem.types';
+import { CardSection } from './CardSection/CardSection';
+import { ICardSectionProps } from './CardSection/CardSection.types';
 
 export const CardView: ICardComponent['view'] = props => {
   const Slots = getSlots<ICardProps, ICardSlots>(props, {
     root: Stack
   });
 
-  const { styles, tokens, compact, ...rest } = props;
+  const { children, styles, tokens, compact, ...rest } = props;
 
   const nativeProps = getNativeProps(rest, htmlElementProperties);
 
+  const CardItemType = (<CardItem /> as React.ReactElement<ICardItemProps>).type;
+  const CardSectionType = (<CardSection /> as React.ReactElement<ICardSectionProps>).type;
+
+  const childrenGap = tokens && (tokens as ICardTokens).childrenGap;
+  const childrenMargin = tokens && (tokens as ICardTokens).childrenMargin;
+  const childrenCount = React.Children.count(children);
+
+  const cardChildren: (React.ReactChild | null)[] = React.Children.map(
+    children,
+    (child: React.ReactElement<ICardItemProps | ICardSectionProps>, index: number) => {
+      if (!child) {
+        return null;
+      }
+
+      let margin: number | string;
+      if (childrenMargin) {
+        const firstMargin: number = index === 0 ? childrenMargin : 0;
+        const lastMargin: number = index === childrenCount - 1 ? childrenMargin : 0;
+
+        const verticalMargin: string = `${firstMargin}px ${childrenMargin}px ${lastMargin}px`;
+        const horizontalMargin: string = `${childrenMargin}px ${lastMargin}px ${childrenMargin}px ${firstMargin}px`;
+
+        margin = compact ? horizontalMargin : verticalMargin;
+      } else {
+        margin = 0;
+      }
+
+      if (child.type === CardItemType || child.type === CardSectionType) {
+        const { fill, tokens: childTokens, ...childRest } = child.props;
+
+        let marginTokens = fill ? { margin: 0 } : { margin };
+
+        if (child.type === CardSectionType) {
+          marginTokens = { ...marginTokens, ...{ childrenGap } };
```

Comment:
I'm still parsing this function and trying to make sense of it. I think a few scattered comments explaining "why" could help here. Why is `childrenGap` only applied to CardSections and not CardItems? Is that intuitive and symmetric with StackItem behavior?

Reply:
`childrenGap` is only applied to `CardSections` because the `childrenGap` token does not exist in `ICardItemTokens`. This is symmetric behavior I think because `childrenGap` applies specifically to `children`, which makes sense since a `CardSection` is made to hold multiple children while `CardItem` is made to hold only one, just like `StackItem`.